barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_to_constraint_buf.cpp

```diff
@@ -105,14 +193,15 @@ Acir::Program deserialize_program(std::vector<uint8_t>&& buf)
             return program;
         },
         &Acir::Program::bincodeDeserialize);
+    BB_ASSERT_EQ(program.functions.size(), 1U, "circuit_buf_to_acir_format: expected single function in ACIR program");
+
+    return circuit_serde_to_acir_format(program.functions[0]);
```

Comment:
Kind of surprised to see this direct [0] access still - do you expect that to go away with the notion of a "program stack" etc.?

Reply:
ok I guess oyu haven't all of that logic in this PR, nvm